pkg/arvo/app/invite-hook.hoon

```diff
@@ -0,0 +1,61 @@
+::  invite-hook: receive invites from any source
+::
+/+  *invite-json
+|%
++$  move  [bone [%poke wire dock [%invite-action invite-action]]]
+--
+::
+|_  [bol=bowl:gall ~]
+::
+++  this  .
+::
+++  poke-json
+  |=  =json
+  ^-  (quip move _this)
+  ?>  (team:title our.bol src.bol)
+  =/  act  (json-to-action json)
+  ?>  ?=(%invite -.act)
+  :_  this
+  [(invite-hook-poke recipient.invite.act act)]~
+::
+++  poke-invite-action
+  |=  act=invite-action
+  ^-  (quip move _this)
+  :_  this
+  ?+  -.act
+    ~
+  ::
+      %invite
+    ?:  (team:title our.bol src.bol)
+      ?>  !(team:title our.bol ship.invite.act)
+      [(invite-hook-poke recipient.invite.act act)]~
+    ?~  (invitatory-scry path.act)  ~
+    ?^  (invite-scry uid.act path.act)  ~
+    [(invite-poke path.act act)]~
+  ==
+::
+++  invite-hook-poke
+  |=  [=ship action=invite-action]
+  ^-  move
+  [ost.bol %poke /invite-hook [ship %invite-hook] [%invite-action action]]
+::
+++  invite-poke
+  |=  [pax=path action=invite-action]
+  ^-  move
+  [ost.bol %poke pax [our.bol %invite-store] [%invite-action action]]
+::
+++  invitatory-scry
+  |=  pax=path
+  ^-  (unit invitatory)
+  =.  pax
+    ;:(weld /=invite-store/(scot %da now.bol)/invitatory pax /noun)
```

Comment:
For wide form, the irregular `:(weld` is more common.

Reply:
I prefer `;:` because it's explicit

Reply:
Could you elaborate on how irregular form is not explicit?

Reply:
Irregular form is nice when you need to save line space or when the expression is nested in such a way that fewer runes adds to readability. In this case, one fewer characters won’t affect the line break, but does make it easier to search in the Docs / etc if someone wants to know what it does. Thus, regular form is slightly more legible to a newcomer.

Reply:
This may all be true, but as a guideline it's not really compatible with hoon. There's tons of irregular syntax, it's used everywhere, and it simply must be learned. There may be some corners of the irregular syntax that can safely be ignored, but `:(foo a bc)` absolutely idiomatic.